CRM/Core/I18n/Schema.php

```diff
@@ -188,18 +188,20 @@ public static function makeSinglelingualTable(
       }
     }
 
+    $dao = new CRM_Core_DAO();
     // deal with columns
     foreach ($columns[$table] as $column => $type) {
-      $queries[] = "ALTER TABLE {$table} ADD {$column} {$type}";
-      $queries[] = "UPDATE {$table} SET {$column} = {$column}_{$retain}";
+      $queries[] = "ALTER TABLE {$table} CHANGE `{$column}_{$retain}` `{$column}` {$type}";
       foreach ($locales as $loc) {
-        $dropQueries[] = "ALTER TABLE {$table} DROP {$column}_{$loc}";
+        if (strcmp($loc,$retain) !== 0) {
```

Comment:
I see now this comment misjudged @mattwire's intent - he's saying we do support 5.1, but we could use `IF EXISTS` more with later MySQL versions to reduce some of that fragility (which I do think is an issue). I'll open that as a separate ticket in JIRA and copy these notes there.